src/modules/krita/producer_krita.c

```diff
@@ -0,0 +1,231 @@
+
+    return r;
+}
+
+static void producer_property_changed(mlt_service owner,
+                                      mlt_producer self,
+                                      mlt_event_data event_data)
+{
+    const char *name = mlt_event_data_to_string(event_data);
+    if (!name)
+        return;
+
+    private_data *pdata = (private_data *) self->child;
+    mlt_properties props = MLT_PRODUCER_PROPERTIES(self);
+
+    if (!strcmp(name, "start_frame") || !strcmp(name, "end_frame")) {
+        pdata->frame_start = mlt_properties_get_int(props, "start_frame");
+        pdata->frame_end = mlt_properties_get_int(props, "end_frame");
+    }
+
+    if (!strcmp(name, "limit_enabled")) {
+        pdata->limit_enabled = mlt_properties_get_int(props, "limit_enabled");
+    }
+
+    if (!strcmp(name, "speed")) {
+        pdata->speed = mlt_properties_get_double(props, "speed");
+    }
+
+    if (!strcmp(name, "_profile")) {
+        mlt_service_set_profile((mlt_service) pdata->producer_internal,
+                                mlt_service_profile((mlt_service) self));
+    }
+}
+
+static void producer_close(mlt_producer producer)
+{
+    private_data *pdata = (private_data *) producer->child;
+
+    if (pdata) {
+        mlt_producer_close(pdata->producer_internal);
+        free(pdata);
+    }
+
+    producer->close = NULL;
+    mlt_producer_close(producer);
+    free(producer);
+}
+
+/** Constructor for the producer.
+*/
+mlt_producer producer_krita_init(mlt_profile profile,
+                                 mlt_service_type type,
+                                 const char *id,
+                                 char *arg)
+{
+    // Create a new producer object
+    mlt_producer producer = mlt_producer_new(profile);
+    private_data *pdata = (private_data *) calloc(1, sizeof(private_data));
+
+    if (arg && producer && pdata) {
+        mlt_properties producer_properties = MLT_PRODUCER_PROPERTIES(producer);
+
+        // Initialize the producer
+        mlt_properties_set(producer_properties, "resource", arg);
+        producer->child = pdata;
+        producer->get_frame = producer_get_frame;
+        producer->seek = producer_seek;
+        producer->close = (mlt_destructor) producer_close;
+
+        // Get the resource to be passed to the clip producer
+        char *resource = arg;
+
+        // Default frame start / end props.
+        pdata->frame_start = -1;
+        pdata->frame_end = -1;
+        pdata->limit_enabled = 1;
+        pdata->speed = 1.0;
+
+        // Initialize property values for start / end frames.
+        mlt_properties_set_int(producer_properties, "start_frame", pdata->frame_start);
+        mlt_properties_set_int(producer_properties, "end_frame", pdata->frame_end);
+        mlt_properties_set_int(producer_properties, "limit_enabled", pdata->limit_enabled);
+        mlt_properties_set_double(producer_properties, "speed", pdata->speed);
+
+        // Create a producer for the clip using the false profile.
```

Comment:
This copied comment is incorrect because this is not using a "false" profile. It uses the same profile as the parent. You could probably just remove the comment to avoid future confusion.